### PR TITLE
KEYCLOAK-9116: Fixes JWK serialization of ECDSA public key coordinates.

### PR DESCRIPTION
**Pull request by larsw**
Signed-off-by: Lars Wilhelmsen <lars@sral.org>

### CODE REVIEW

**Comment by pedroigor** [comment]
@keycloak-ci-bot test

File: core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java [hunk 1]
```diff
@@ -77,24 +77,108 @@ public JWK rsa(Key key) {
...
+        int fieldSize = ecKey.getParams().getCurve().getField().getFieldSize();
+        BigInteger affineX = ecKey.getW().getAffineX();
+        BigInteger affineY = ecKey.getW().getAffineY();
+
         k.setKeyId(kid);
         k.setKeyType(KeyType.EC);
         k.setAlgorithm(algorithm);
         k.setPublicKeyUse(DEFAULT_PUBLIC_KEY_USE);
-        k.setCrv("P-" + ecKey.getParams().getCurve().getField().getFieldSize());
-        k.setX(Base64Url.encode(ecKey.getW().getAffineX().toByteArray()));
```
Comment (pedroigor) [inline core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java]: I think you can fix this issue by just using the `toIntegerBytes` method. Something like:

```java
k.setX(Base64Url.encode(toIntegerBytes(ecKey.getW().getAffineX())));
k.setY(Base64Url.encode(toIntegerBytes(ecKey.getW().getAffineY())));
```

That method skips the signbit.

**Review by pedroigor — CHANGES_REQUESTED** [review]

**Review by abstractj — CHANGES_REQUESTED** [review]
@larsw could you please update your PR based on Pedro's suggestion keeping the test you added? I believe that's a better alternative to fix this issue.

File: core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java (hunk 1 above)
Comment (larsw) [reply core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java]: Done.

**Comment by pedroigor** [comment]
@larsw, tks. You only need to squash your commits and we are ready to go as soon as CI is happy. Please, keep the message `KEYCLOAK-9116: Fixes JWK serialization of ECDSA public key coordinates.` after squashing the commits.

**Comment by abstractj** [comment]
@larsw could you please squash your changes in a single commit?

**Comment by larsw** [comment]
> @larsw, tks. You only need to squash your commits and we are ready to go as soon as CI is happy. Please, keep the message `KEYCLOAK-9116: Fixes JWK serialization of ECDSA public key coordinates.` after squashing the commits.

Done.

**Comment by abstractj** [comment]
@keycloak-ci-bot test

**Review by abstractj — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

**Comment by fredbi** [comment]
:clap:

**Pull request merged**